lib/checkother.cpp

```diff
@@ -2700,7 +2706,7 @@ void CheckOther::checkDuplicateExpression()
                                     while (parent && parent->astParent()) {
                                         parent = parent->astParent();
                                     }
-                                    if (parent && parent->previous() && parent->strAt(-1) == "static_assert") {
+                                    if (parent && parent->previous() && isStaticAssert(parent->previous())) {
```

Comment:
hmm.. technically the `&& parent->previous()` is redundant. I could live with a redundant condition but I prefer that it's removed. `Token::Match` is well-known to check the parameter imho.